client/src/DataLayer.tsx

```diff
@@ -202,8 +203,9 @@ export class DataLayer extends React.Component<DataLayerProps, DataLayerState> {
 
     if (allItems === undefined || allLists === undefined) {
       if (!this.state.isLoadingFresh) {
-        this.loadData(HnListSource.Front);
+        console.log("local storage is empty, loading fresh data based on active page", this.props.loadFreshSource);
         this.setState({ isLoadingFresh: true });
+        this.loadData(this.props.loadFreshSource);
```

Comment:
I deliberately moved the call to `loadData` to happen after the call to `setState`. This seemed appropriate to me, since it should reduce the chance that multiple calls to `loadData` are allowed (race condition on `isLoadingFresh`).